### PR TITLE
basic dataset

### PR DESCRIPTION
**Pull request by clee-ai**
@humanpose1 let me know what you think... it's very crude but it at least runs. 

Right now it will instantiate the s3dis1x1 dataset.

You can try running it with `python train.py model=default dataset=segmentation/s3dis/s3dis1x1.yaml`

I think the first thing we should figure out is how to properly implement static typechecking for the configs (#4) or just abandon the `DataConfig`. It looks like pytorch lightning transformers implemented the dataconfig but didn't actually make it do any type checking...

### CODE REVIEW

**Comment by clee-ai** [comment]
It looks like what we'd need to do is convert the dictconf into the dataclasses using https://omegaconf.readthedocs.io/en/latest/usage.html#omegaconf-to-object ?

File: torch_points3d/model/base_model.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+import pytorch_lightning as pl
+import torch
+from pytorch_lightning.utilities import rank_zero_info
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+from torch_points3d.core.instantiator import Instantiator
+
+
+@dataclass
```
Comment (humanpose1) [inline torch_points3d/model/base_model.py]: repetition. in `torch_points3d/core/config.py` there are already these dataclasses

File: conf/config.yaml [hunk 2]
```diff
@@ -1,19 +1,12 @@
-defaults:
-  - task: ???
-  - models: ???
-  - data: ???
+defaults: # loads default configs
```
Comment (humanpose1) [inline conf/config.yaml:1]: Ok for now. Maybe in the future, we will change this to have multiple optimizers.

File: torch_points3d/model/base_model.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    def configure_metrics(self, stage: str) -> Optional[Any]:
+        """
+        Override to configure metrics for train/validation/test.
+        This is called on fit start to have access to the data module,
+        and initialize any data specific metrics.
+        """
+
+    def on_save_checkpoint(self, checkpoint: Dict[str, Any]):
+        # Save tokenizer from datamodule for predictions
```
Comment (humanpose1) [inline torch_points3d/model/base_model.py]: is this method useful in our case ?

**Review by humanpose1 — COMMENTED** [review]
Thanks for this PR. It cool that it can run !I've made small remarks.

For optimizer I don't understand why there is a duplicate.


About the type checking. How can we do type checking for data augmentation with hydra dataclasses ?

**Comment by humanpose1** [comment]
Btw I got this error:
```
TypeError: Error instantiating 'torch_points3d.dataset.s3dis1x1.s3dis_data_module' : 'DictConfig' object is not callable

```

File: conf/dataset/default.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+# cfg:
+  # torch data-loader specific arguments
+cfg: 
+  batch_size: ${training.batch_size}
+  num_workers: ${training.num_workers}
+  dataroot: data
+
+  common_transform: {}
+  aug_transform: {}
+  pre_transform: {}
```
Comment (humanpose1) [inline conf/dataset/default.yaml]: Maybe put None here instead of dict

File: conf/config.yaml (hunk 2 above)
Comment (clee-ai) [reply conf/config.yaml:1]: I agree, eventually we should change the optimizer to be part of the `model` package.

File: torch_points3d/model/base_model.py (hunk 3 above)
Comment (clee-ai) [reply torch_points3d/model/base_model.py]: Hmm, I'm guessing not since the Initializer won't change between runs.

**Comment by clee-ai** [comment]
That's strange about the error, can you send a stack trace?

**Pull request closed without merging**